core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -0,0 +1,1508 @@
+        isCurrentlyActive: Boolean
+      ) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+        viewHolder.itemView.alpha = 1 - abs(dY) / viewHolder.itemView.measuredHeight
+      }
+
+      override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        target: RecyclerView.ViewHolder
+      ): Boolean = false
+
+      override fun onSwiped(
+        viewHolder: RecyclerView.ViewHolder,
+        direction: Int
+      ) {
+        closeTab(viewHolder.adapterPosition)
+      }
+    }
+  }
+
+  private fun initHideBackToTopTimer() {
+    hideBackToTopTimer = object : CountDownTimer(1200, 1200) {
+      override fun onTick(millisUntilFinished: Long) {}
+      override fun onFinish() {
+        activity_main_back_to_top_fab?.hide()
+      }
+    }
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? = inflater.inflate(R.layout.fragment_reader, container, false)
+
+  private fun handleIntentExtras(intent: Intent) {
+    if (intent.hasExtra(TAG_FILE_SEARCHED)) {
+      val openInNewTab = (isInTabSwitcher ||
+        intent.getBooleanExtra(TAG_FILE_SEARCHED_NEW_TAB, false))
+      searchForTitle(
+        intent.getStringExtra(TAG_FILE_SEARCHED),
+        openInNewTab
+      )
+      selectTab(webViewList.size - 1)
+    }
+    handleNotificationIntent(intent)
+  }
+
+  private val isInTabSwitcher: Boolean
+    get() = mainMenu != null && mainMenu!!.isInTabSwitcher()
+
+  private fun handleNotificationIntent(intent: Intent) {
+    if (intent.hasExtra(DOWNLOAD_NOTIFICATION_TITLE)) {
+      Handler().postDelayed(
+        {
+          val bookMatchingTitle =
+            newBookDao.bookMatching(intent.getStringExtra(DOWNLOAD_NOTIFICATION_TITLE))
+          if (bookMatchingTitle != null) {
+            openZimFile(bookMatchingTitle.file)
+          }
+        },
+        300
+      )
+    }
+  }
+
+  private fun setupDocumentParser() {
+    documentParser = DocumentParser(object : SectionsListener {
+      override fun sectionsLoaded(
+        title: String,
+        sections: List<DocumentSection>
+      ) {
+        if (isAdded) {
+          documentSections!!.addAll(sections)
+          tableDrawerAdapter!!.setTitle(title)
+          tableDrawerAdapter!!.setSections(documentSections!!)
+          tableDrawerAdapter!!.notifyDataSetChanged()
+        }
+      }
+
+      override fun clearSections() {
+        documentSections!!.clear()
+        if (tableDrawerAdapter != null) {
+          tableDrawerAdapter!!.notifyDataSetChanged()
+        }
+      }
+    })
+  }
+
+  private fun setTabListener() {
+    tabsAdapter!!.setTabClickListener(object :
+      TabsAdapter.TabClickListener {
+      override fun onSelectTab(view: View, position: Int) {
+        hideTabSwitcher()
+        selectTab(position)
+
+        /* Bug Fix #592 */updateBottomToolbarArrowsAlpha()
```

Comment:
Can we move the method call below the comment or append the comment after call. I guess first one would look nicer